### DISCUSSION TITLE
Differ Switch and Checkbox

### DISCUSSION THREAD

**Discussion by Kuna42** [beeware/toga#2221] [Ideas]
What if both exists, Switch and Checkbox, and platforms which do not support Checkbox by native, have a Switch instead.

Sure, it is bad, that you have two different classes which maps to the same widget layout on some platforms, but i think it is better instead of shrinking different kinds of widgets.

**Comment by freakboy3742** [comment]
I've have similar thoughts recently. 

Switch was one of the first widgets I added to Toga; when I wrote it, the theory was that platforms all have a single platform-appropriate way of representing Boolean properties. I picked "Switch" specifically to avoid using "checkbox" to highlight the difference in how most GUI toolkits represent the concept.

However, in the intervening years, platforms have gotten a bit loose about this. Even macOS has started including both switches and checkboxes in their GUIs.

I've opened #2225 to track this idea - thanks for the suggestion!
